src/Http/WebUtilities/src/HttpRequestStreamReader.cs

```diff
@@ -178,17 +191,19 @@ public override int Read(char[] buffer, int index, int count)
                     charsRemaining = count;
                 }
 
-                Buffer.BlockCopy(
```

Comment:
I'd be curious about the perf difference between Buffer.BlockCopy vs ReadOnlySpan.CopyTo? @benaadams @davidfowl my gut says Span.CopyTo(Span) would be faster.

Reply:
I think `Span`.`CopyTo` is not worse than `BlockCopy`.

I wrote a quick benchmark [here](https://github.com/alefranz/AspNetCore/blob/benchmarks-PR-18802/src/Http/WebUtilities/perf/Microsoft.AspNetCore.WebUtilities.Performance/HttpRequestStreamReaderBenchmark.cs), which confirm that with only one exception but I believe it's more due to my mobile CPU affected by power or temperature limit (as similar tests went from 61ms to 60ms)

|     Method |     Toolchain | RunStrategy |    Size |  StartIndex |          Mean |         Error |        StdDev |          Op/s | Allocated |
| ---------- |-------------- |------------ |-------- |------- |--------------:|--------------:|--------------:|--------------:|----------:|
|  SpanCopy | .NET Core 5.0 |  Throughput |      10 |      0 |      2.866 ns |     0.0159 ns |     0.0141 ns | 348,882,333.8 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput |      10 |      0 |     12.379 ns |     0.0541 ns |     0.0452 ns |  80,784,172.6 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput |      10 |      5 |      3.800 ns |     0.0631 ns |     0.0590 ns | 263,176,907.7 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput |      10 |      5 |     12.117 ns |     0.0733 ns |     0.0685 ns |  82,529,330.1 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput |    1000 |      0 |     32.270 ns |     0.2395 ns |     0.2123 ns |  30,988,729.4 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput |    1000 |      0 |     40.379 ns |     0.3587 ns |     0.3356 ns |  24,765,399.0 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput |    1000 |      5 |     32.146 ns |     0.2466 ns |     0.2307 ns |  31,108,259.3 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput |    1000 |      5 |     40.839 ns |     0.3113 ns |     0.2912 ns |  24,486,621.9 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput |    1000 |     50 |     30.439 ns |     0.1588 ns |     0.1485 ns |  32,852,701.4 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput |    1000 |     50 |     38.509 ns |     0.3944 ns |     0.3689 ns |  25,968,039.3 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput | 1000000 |      0 | 61,305.950 ns |   682.7159 ns |   638.6128 ns |      16,311.6 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput | 1000000 |      0 | 61,508.315 ns | 1,152.1334 ns | 1,077.7062 ns |      16,258.0 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput | 1000000 |      5 | **61,298.624 ns** |   452.6076 ns |   423.3694 ns |      16,313.6 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput | 1000000 |      5 | 60,571.077 ns |   332.9048 ns |   311.3993 ns |      16,509.5 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput | 1000000 |     50 | 60,371.150 ns |   365.6293 ns |   342.0098 ns |      16,564.2 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput | 1000000 |     50 | 60,837.349 ns |   390.9090 ns |   365.6565 ns |      16,437.3 |       0 B |
|  SpanCopy | .NET Core 5.0 |  Throughput | 1000000 | 345678 | 39,506.150 ns |   257.0060 ns |   214.6116 ns |      25,312.5 |       0 B |
| BlockCopy | .NET Core 5.0 |  Throughput | 1000000 | 345678 | 39,729.555 ns |   272.2552 ns |   241.3471 ns |      25,170.2 |       0 B |

However I imagine there are better data on this somewhere, but I was not able to find it.

Reply:
This is plenty enough data; I think we are fine to use Span.CopyTo here.